### PR TITLE
#1959 - Staging sims prod environment : formio GitHub action build deploy split

### PR DESCRIPTION
**Pull request by guru-aot**
Split the github action workflow which currently builds and deploys simultaneously in a single action into seperate build and deploy github action.

### CODE REVIEW

File: .github/workflows/env-setup-build-forms-server.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        default: ""
+
+jobs:
+  build:
+    name: Build Forms Server
+    runs-on: ubuntu-latest
+    env:
+      FORMIO_SOURCE_REPO_TAG: ${{ inputs.formioTag }}
+      BUILD_REF: ${{ inputs.formioTag }}
+      BUILD_NAMESPACE: 0c27fb-tools
```
Comment (andrewsignori-aot) [inline .github/workflows/env-setup-build-forms-server.yml:24]: We have an env variable for this now.

File: .github/workflows/env-setup-build-forms-server.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        run: |
+          echo BUILD NAMESPACE: $BUILD_NAMESPACE
+          echo BRANCH: $BUILD_REF
+      # Checkout the PR branch
+      - name: Checkout Target Branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.gitRef }}
+      # Log in to OpenShift.
+      # Note: The secrets needed to log in are NOT available if the PR comes from a FORK.
```
Comment (andrewsignori-aot) [inline .github/workflows/env-setup-build-forms-server.yml:36]: I do not believe that comments on line 36 and 37 ate necessary n our context.

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, minor comments only.

**Review by sh16011993 — CHANGES_REQUESTED** [review]
Nice work 👍 . Please have a look at @andrewsignori-aot  comments.

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍

**Review by dheepak-aot — APPROVED** [review]

**Review by sh16011993 — APPROVED** [review]

File: .github/workflows/env-setup-build-forms-server.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    inputs:
+      formioTag:
+        description: "Formio tag, please refer https://github.com/formio/formio/tags to deploy the appropriate tag."
+        type: string
+        required: true
+        default: "v2.5.3"
+      gitRef:
+        description: "Git Ref"
+        required: true
+        default: ""
```
Comment (ann-aot) [inline .github/workflows/env-setup-build-forms-server.yml:15]: why should we need the default as empty?

File: .github/workflows/env-setup-build-forms-server.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    env:
+      FORMIO_SOURCE_REPO_TAG: ${{ inputs.formioTag }}
+      BUILD_REF: ${{ inputs.formioTag }}
+      BUILD_NAMESPACE: ${{ vars.BUILD_NAMESPACE }}
+    steps:
+      - name: Print env
+        run: |
+          echo BUILD NAMESPACE: $BUILD_NAMESPACE
+          echo BRANCH: $BUILD_REF
+      # Checkout the PR branch
```
Comment (ann-aot) [inline .github/workflows/env-setup-build-forms-server.yml:30]: period

Comment (ann-aot) [reply .github/workflows/env-setup-build-forms-server.yml:30]: line 39 too

Comment (guru-aot) [reply .github/workflows/env-setup-build-forms-server.yml:30]: Will update it in my next pr

File: .github/workflows/env-setup-build-forms-server.yml (hunk 3 above)
Comment (guru-aot) [reply .github/workflows/env-setup-build-forms-server.yml:15]: The git ref for build is the reference something like main-65, it does not have any default value.

**Pull request merged**